mi/dataset/driver/zplsc_c/zplsc_c_echogram.py

```diff
@@ -0,0 +1,300 @@
+        # rotates and right aligns the x labels, and moves the bottom of the
+        # axes up to make room for them
+        ax.set_xlabel('time (UTC)', fontsize=ZPLSCCPlot.font_size_small)
+        ax.set_xticks(xticks)
+        ax.set_xticklabels(xticklabels, rotation=45, horizontalalignment='center', fontsize=ZPLSCCPlot.font_size_small)
+        ax.set_xlim(0, time_length)
+
+    @staticmethod
+    def _display_colorbar(fig, data_axes):
+        # Add a colorbar to the specified figure using the data from the given axes
+        ax = fig.add_axes([0.965, 0.12, 0.01, 0.775])
+        cb = fig.colorbar(data_axes, cax=ax, use_gridspec=True)
+        cb.set_label('dB', fontsize=ZPLSCCPlot.font_size_large)
+        cb.ax.tick_params(labelsize=ZPLSCCPlot.font_size_small)
+
+
+class ZPLSCCEchogram(object):
+    def __init__(self):
+        self.cc = ZplscCCalibrationCoefficients()
+        self.params = ZplscCParameters()
+
+    def compute_backscatter(self, profile_hdr, chan_data, sound_speed, depth_range, sea_absorb):
+        """
+        Compute the backscatter volumes values for one zplsc_c profile data record.
+        This code was borrowed from ASL MatLab code that reads in zplsc-c raw data
+        and performs calculations in order to compute the backscatter volume in db.
+
+        :param profile_hdr: Raw profile header with metadata from the zplsc-c instrument.
+        :param chan_data: Raw frequency data from the zplsc-c instrument.
+        :param sound_speed: Speed of sound at based on speed of sound, pressure and salinity.
+        :param depth_range: Range of the depth of the measurements
+        :param sea_absorb: Seawater absorption coefficient for each frequency
+        :return: sv: Volume backscatter in db
+        """
+
+        _N = []
+        if self.params.Bins2Avg > 1:
+            for chan in range(profile_hdr.num_channels):
+                el = self.cc.EL[chan] - 2.5/self.cc.DS[chan] + np.array(chan_data[chan])/(26214*self.cc.DS[chan])
+                power = 10**(el/10)
+
+                # Perform bin averaging
+                num_bins = len(chan_data[chan])/self.params.Bins2Avg
+                pwr_avg = []
+                for _bin in range(num_bins):
+                    pwr_avg.append(np.mean(power[_bin*self.params.Bins2Avg:(_bin+1)*self.params.Bins2Avg]))
+
+                el_avg = 10*np.log10(pwr_avg)
+                _N.append(np.round(26214*self.cc.DS[chan]*(el_avg - self.cc.EL[chan] + 2.5/self.cc.DS[chan])))
+
+        else:
+            for chan in range(profile_hdr.num_channels):
+                _N.append(np.array(chan_data[chan]))
+
+        sv = []
+        for chan in range(profile_hdr.num_channels):
+            # Calculate correction to Sv due to non square transmit pulse
+            sv_offset = zf.compute_sv_offset(profile_hdr.frequency[chan], profile_hdr.pulse_length[chan])
+            sv.append(self.cc.EL[chan]-2.5/self.cc.DS[chan] + _N[chan]/(26214*self.cc.DS[chan]) - self.cc.TVR[chan] -
```

Comment:
place numbers like 26214 and 2.5 into named variables to better explain what they are (if you know what they are)

Reply:
I totally agree.  If you saw code that I wrote, my variable and constant names would be so long you would know exactly what they are.  Unfortunately, that code os from the ASL MatLab code that had no comments.